File: Swap-Closet-V4/backend/src/main/java/org/swapcloset/backend/dto/UsuarioDTO.java
package org.swapcloset.backend.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.*;

import java.util.List;

@Data
public class UsuarioDTO {
    private Integer id;
    @NotNull(message = "El nombre ni puede ser nulo")
    @NotBlank(message = "El nombre no puede estar vacío")
    private String nombre;
    @NotNull(message = "Los apellidos no pueden ser nulos")
    @NotBlank(message = "Los apellidos no pueden estar vacíos")
    private String apellidos;
    @Email(message = "El email no es válido")
    private String email;
    @NotNull(message = "La contraseña no puede ser nula")
    @NotBlank(message = "La contraseña no puede estar vacía")
    private String password;
    private String descripcion;
    private String estilo;
    private String urlImg;
    private String direccion;
    @JsonProperty("tCamiseta")
    private String tCamiseta;
    @JsonProperty("tPantalon")
    private Integer tPantalon;
    @JsonProperty("tCalzado")
    private Integer tCalzado;

    private List<ProductoDTO> productos;
}

File: Swap-Closet-V4/frontend/src/app/modelos/UsuarioDTO.ts
import {RaitingDTO} from "./RaitingDTO";
import {ProductoDTO} from "./ProductoDTO";
import {ChatDTO} from "./ChatDTO";

export interface UsuarioDTO {

  id?: number
  nombre?: string
  apellidos?: string
  email?: string
  password?: string
  descripcion?: string
  estilo?: string
  urlImg?: string
  direccion?: string
  tCamiseta?: string
  tPantalon?: number
  tCalzado?: number

  productos?: ProductoDTO[];
}
